pycgm/pycgm.py

```diff
@@ -0,0 +1,5263 @@
+            `point` is a point not on the line.
+            `start` and `end` form a line.
+
+        Returns
+        -------
+        dist, nearest, point : tuple
+            `dist` is the closest distance from the point to the line.
+            `nearest` is the closest point on the line from `point`.
+            It is represented as a 1x3 array.
+            `point` is the original point not on the line.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from .pycgm import CGM
+        >>> point = np.array([1, 2, 3])
+        >>> start = np.array([4, 5, 6])
+        >>> end = np.array([7, 8, 9])
+        >>> dist, nearest, point = CGM.point_to_line(point, start, end)
+        >>> np.around(dist, 8)
+        5.19615242
+        >>> np.around(nearest, 8)
+        array([4., 5., 6.])
+        >>> np.around(point, 8)
+        array([1, 2, 3])
+        """
+        line_vector = end - start
+        point_vector = point - start
+        line_length = np.linalg.norm(line_vector)
+        line_unit_vector = line_vector / line_length
+        point_vector_scaled = point_vector * (1.0 / line_length)
+        t = np.dot(line_unit_vector, point_vector_scaled)
+        if t < 0.0:
+            t = 0.0
+        elif t > 1.0:
+            t = 1.0
+
+        nearest = line_vector * t
+        dist = np.linalg.norm(point_vector - nearest)
+        nearest = nearest + start
+
+        return dist, nearest, point
+
+    @staticmethod
+    def find_l5(lhjc, rhjc, axis):
+        """Estimates the L5 marker position given the pelvis or thorax axis.
+
+        Markers used : LHJC, RHJC
+
+        Parameters
+        ----------
+        lhjc, rhjc : ndarray
+            1x3 ndarray giving the XYZ coordinates of the LHJC and RHJC
+            markers respectively.
+        axis : ndarray
+            Numpy array containing 4 1x3 arrays of pelvis or thorax origin, x-axis, y-axis,
+            and z-axis. Only the z-axis affects the estimated L5 result.
+
+        Returns
+        -------
+        mid_hip, l5 : tuple
+            `mid_hip` is a 1x3 ndarray giving the XYZ coordinates of the middle
+            of the LHJC and RHJC markers. `l5` is a 1x3 ndarray giving the estimated
+            XYZ coordinates of the L5 marker.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from .pycgm import CGM
+        >>> lhjc = np.array([308, 322, 937])
+        >>> rhjc = np.array([182, 339, 935])
+        >>> axis = np.array([[251, 391, 1032],
+        ...                  [251, 392, 1032],
+        ...                  [250, 391, 1032],
+        ...                  [251, 391, 1033]])
+        >>> np.around(CGM.find_l5(lhjc, rhjc, axis), 8)
+        array([[ 245.        ,  330.5       ,  936.        ],
+               [ 271.06445299,  371.10239489, 1043.26924277]])
+        """
+        # The L5 position is estimated as (LHJC + RHJC)/2 +
```

Comment:
This is a pretty good comment that seems like useful information for a docstring